src/imitation/algorithms/preference_comparisons.py

```diff
@@ -509,7 +509,8 @@ def probability(self, rews1: th.Tensor, rews2: th.Tensor) -> th.Tensor:
         if self.discount_factor == 1:
             returns_diff = (rews2 - rews1).sum(axis=0)  # type: ignore[call-overload]
         else:
-            discounts = self.discount_factor ** th.arange(len(rews1))
+            device = rews1.device
```

Comment:
We could have problems if `rews1` and `rews2` are on different devices, I think? That shouldn't happen, but maybe worth adding an explicit assert to check they're the same?

Reply:
Fixed